tracing-subscriber/src/filter/env/mod.rs

```diff
@@ -484,6 +476,86 @@ impl<S: Subscriber> Layer<S> for EnvFilter {
     }
 }
 
+impl<S: Subscriber> Layer<S> for EnvFilter {
+    #[inline]
+    fn register_callsite(&self, metadata: &'static Metadata<'static>) -> Interest {
+        EnvFilter::register_callsite(self, metadata)
+    }
+
+    #[inline]
+    fn max_level_hint(&self) -> Option<LevelFilter> {
+        EnvFilter::max_level_hint(self)
+    }
+
+    #[inline]
+    fn enabled(&self, metadata: &Metadata<'_>, _: Context<'_, S>) -> bool {
+        self.enabled(metadata)
+    }
+
+    #[inline]
+    fn on_new_span(&self, attrs: &span::Attributes<'_>, id: &span::Id, _: Context<'_, S>) {
+        self.on_new_span(attrs, id)
+    }
+
+    fn on_record(&self, id: &span::Id, values: &span::Record<'_>, _: Context<'_, S>) {
+        if let Some(span) = try_lock!(self.by_id.read()).get(id) {
+            span.record_update(values);
+        }
+    }
+
+    #[inline]
+    fn on_enter(&self, id: &span::Id, _: Context<'_, S>) {
+        self.on_enter(id);
+    }
+
+    #[inline]
+    fn on_exit(&self, id: &span::Id, _: Context<'_, S>) {
+        self.on_exit(id);
+    }
+
+    #[inline]
+    fn on_close(&self, id: span::Id, _: Context<'_, S>) {
+        self.on_close(id);
+    }
+}
+
+impl<S> layer::Filter<S> for EnvFilter {
+    #[inline]
+    fn enabled(&self, meta: &Metadata<'_>, _: &Context<'_, S>) -> bool {
+        self.enabled(meta)
+    }
+
+    #[inline]
+    fn callsite_enabled(&self, meta: &'static Metadata<'static>) -> Interest {
+        self.register_callsite(meta)
+    }
+
+    #[inline]
+    fn max_level_hint(&self) -> Option<LevelFilter> {
+        EnvFilter::max_level_hint(self)
+    }
+
+    #[inline]
+    fn on_new_span(&self, attrs: &span::Attributes<'_>, id: &span::Id, _: Context<'_, S>) {
+        self.on_new_span(attrs, id)
+    }
+
+    #[inline]
+    fn on_enter(&self, id: &span::Id, _: Context<'_, S>) {
+        self.on_enter(id);
+    }
+
+    #[inline]
+    fn on_exit(&self, id: &span::Id, _: Context<'_, S>) {
+        self.on_exit(id);
+    }
+
+    #[inline]
+    fn on_close(&self, id: span::Id, _: Context<'_, S>) {
+        self.on_close(id);
+    }
+}
```

Comment:
The `Filter` trait depends on the "registry" feature flag, so this is failing CI: https://github.com/tokio-rs/tracing/runs/5499039977?check_suite_focus=true

I would wrap this in the `feature!` macro:

```suggestion
feature! {
    #![all(feature = "registry", feature = "std")]
    
    impl<S> layer::Filter<S> for EnvFilter {
        #[inline]
        fn enabled(&self, meta: &Metadata<'_>, _: &Context<'_, S>) -> bool {
            self.enabled(meta)
        }

        #[inline]
        fn callsite_enabled(&self, meta: &'static Metadata<'static>) -> Interest {
            self.register_callsite(meta)
        }

        #[inline]
        fn max_level_hint(&self) -> Option<LevelFilter> {
            EnvFilter::max_level_hint(self)
        }

        #[inline]
        fn on_new_span(&self, attrs: &span::Attributes<'_>, id: &span::Id, _: Context<'_, S>) {
            self.on_new_span(attrs, id)
        }

        #[inline]
        fn on_enter(&self, id: &span::Id, _: Context<'_, S>) {
            self.on_enter(id);
        }

        #[inline]
        fn on_exit(&self, id: &span::Id, _: Context<'_, S>) {
            self.on_exit(id);
        }

        #[inline]
        fn on_close(&self, id: span::Id, _: Context<'_, S>) {
            self.on_close(id);
        }
    }
}
```